From: J. Bruce Fields <bfields@citi.umich.edu>
Subject: nfs4 acl listattr fix
Patch-mainline: yes
References: 53289

Thanks to Frank Filz for pointing out that we list system.nfs4_acl extended
attribute even on filesystems where we don't actually support nfs4_acl.
This is inconsistent with the e.g. ext3 POSIX ACL behaviour, and seems to
annoy cp.

Signed-off-by: J. Bruce Fields <bfields@citi.umich.edu>
Acked-by: NeilBrown <neilb@suse.de>

Index: linux-2.6.15/fs/nfs/nfs4proc.c
===================================================================
--- linux-2.6.15.orig/fs/nfs/nfs4proc.c	2006-02-27 12:49:08.000000000 +1100
+++ linux-2.6.15/fs/nfs/nfs4proc.c	2006-02-27 12:54:13.000000000 +1100
@@ -3588,6 +3588,8 @@
 {
 	size_t len = strlen(XATTR_NAME_NFSV4_ACL) + 1;
 
+	if (!nfs4_server_supports_acls(NFS_SERVER(dentry->d_inode)))
+		return 0;
 	if (buf && buflen < len)
 		return -ERANGE;
 	if (buf)
